### PR TITLE
Api

### PR DESCRIPTION
**Pull request by vdegans**
* Hash passwords before comparing with the database
* Minor lint fixes
* JWT configuration changes
* Add register endpoint
* Fix regex check for passwords

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -13,11 +13,19 @@
...
 builder.Services.AddScoped<MemberAuthorization>();
 builder.Services.AddScoped<MemberRegistration>();
-builder.Services.AddAuthentication().AddJwtBearer(options => options.TokenValidationParameters = new TokenValidationParameters
-{
-    ValidIssuer = builder.Configuration["Jwt:Issuer"],
-    ValidAudience = builder.Configuration["Jwt:Audience"],
-    IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(builder.Configuration["Jwt:Key"]))
+
+builder.Services.AddAuthentication().AddJwtBearer(options => {
+    options.SaveToken = true;
```
Comment (Ovvugt) [inline Program.cs]: SaveToken seems redundant as we do not really on any of its functionalities

File: Controllers/RegisterController.cs [hunk 2]
```diff
@@ -15,8 +13,8 @@ public RegisterController(MemberRegistration memberRegistration)
         _memberRegistration = memberRegistration;
     }
 
-    [HttpPost]
-    public async Task<IActionResult> Register(RegisterDTO userDetails)
+    [HttpPost("api/register")]
+    public async Task<IActionResult> Register([FromBody] RegisterDTO userDetails)
```
Comment (Ovvugt) [inline Controllers/RegisterController.cs]: Why the FromBody attribute, isn't this the default for Post methods?

File: Controllers/RegisterController.cs [hunk 3]
```diff
@@ -15,8 +13,8 @@ public RegisterController(MemberRegistration memberRegistration)
         _memberRegistration = memberRegistration;
     }
 
-    [HttpPost]
-    public async Task<IActionResult> Register(RegisterDTO userDetails)
+    [HttpPost("api/register")]
```
Comment (Ovvugt) [inline Controllers/RegisterController.cs]: This attribute can be changed back to default, and [Route("api/[controller]")] and [ApiController] Attributes should be added to the controller class

**Review by Ovvugt — CHANGES_REQUESTED** [review]
A few comments where added to specific files, asides from these minor details good work! :)

File: Program.cs (hunk 1 above)
Comment (vdegans) [reply Program.cs]: agreed, removed

**Review by Ovvugt — APPROVED** [review]
All good now

**Pull request merged**